src/normalwishart.jl

```diff
@@ -62,27 +60,28 @@ function logpdf(nw::NormalWishart, x::Vector{T}, Lam::Matrix{T}) where T<:Real
         hp = 0.5 * p
 
         # Normalization
-        logp = hp*(log(kappa) - Float64(log2π))
+        logp = hp*(log(kappa) - Float64(log(2*π)))
         logp -= hnu * logdet(Tchol)
         logp -= hnu * p * log(2.)
-        logp -= lpgamma(p, hnu)
+        logp -= logmvgamma(p, hnu)
 
         # Wishart (MvNormal contributes 0.5 as well)
         logp += (hnu - hp) * logdet(Lam)
-        logp -= 0.5 * trace(Tchol \ Lam)
+        T0 = Tchol[:U]'*Tchol[:U]
+        logp -= 0.5 * trace(T0 * Lam)
 
         # Normal
         z = nw.zeromean ? x : x - mu
         logp -= 0.5 * kappa * dot(z, Lam * z)
 
         return logp
-
     end
 end
 
 function rand(nw::NormalWishart)
-    Lam = rand(Wishart(nw.nu, nw.Tchol))
-    Lsym = PDMat(Symmetric(inv(Lam) ./ nw.kappa))
-    mu = rand(MvNormal(nw.mu, Lsym))
+    V = PDMat(Symmetric(inv(nw.Tchol)))
```

Comment:
Yes. Julia's matrix inversion can lead to some small inaccuracies (of ordr 1e-16 or 1e-18 or so) and thus inverse matrices that SHOULD be symmetric will be slightly asymmetric. This shouldn't be a problem, but the distributions we are passing these two require symmetric matrices, and enforce a check for symmetricity. Julia's base is_hermitian function does not have a relative tolerance, and thus without some way of enforcing that the inverted matrix is symmetric, this will crash.

Reply:
We discussed this somewhere, and this was the conclusion, but I cannot find it anymore.

Reply:
I suppose this might have something to do with the missing `isApproxSymmetric` function...

Reply:
The problem is that the symmetry checks are not in the ConjugatePriors package, they are in various places in Distributions. So even if we make our symmetry checks sensitive to a relative tolerance, the checks there won't be. To be honest I have no idea why Julia won't just add a relative tolerance to its is_hermitian function given that the symmetry errors are coming from Julia's own matrix inversion routines, but I was told that that isn't going to happen...